src/levenberg_marquardt.jl

```diff
@@ -0,0 +1,170 @@
+immutable LevenbergMarquardt <: Optim.Optimizer end
```

Comment:
Looking at the code, I guess this is subtyped because https://github.com/JuliaOpt/Optim.jl/blob/c7333f53da8632d60645f42bee27507ae3a78a0b/src/types.jl#L48 is unnecessarily type annotated. I mean, that could just as well be `T` without the `<: Optimizer` in that line.

I will remove it in the next tag of Optim, but maybe let it stay until then, such that we don't have to copy all of the trace code here. Is that :+1: ?